cocos/game/game.ts

```diff
@@ -770,20 +770,8 @@ export class Game extends EventTarget {
                 return deviceManager.init(this.canvas, bindingMappingInfo);
             })
             .then(() => {
-                const renderPipelineUuid = settings.querySettings(Settings.Category.RENDERING, 'renderPipeline') as string;
-                if (!renderPipelineUuid) {
-                    // Editor, noop
-                } else if (renderPipelineUuid === 'ca127c79-69d6-4afd-8183-d712d7b80e14') { // builtin-pipeline
-                    // if custom-pipeline is not feature cropped and macro.CUSTOM_PIPELINE_NAME is not set
-                    if (cclegacy.rendering && !macro.CUSTOM_PIPELINE_NAME) {
-                        // set macro.CUSTOM_PIPELINE_NAME to Builtin
-                        macro.CUSTOM_PIPELINE_NAME = 'Builtin';
-                    }
-                }
-                // if custom-pipeline is not feature cropped and macro.CUSTOM_PIPELINE_NAME is not set
-                // Use legacy render pipeline
-                if (macro.CUSTOM_PIPELINE_NAME === '') {
-                    cclegacy.rendering = undefined;
+                if (cclegacy.rendering && !macro.CUSTOM_PIPELINE_NAME) {
```

Comment:
Why need to check `cclegacy.rendering`?

Reply:
`macro.CUSTOM_PIPELINE_NAME` is only used by custom pipeline.
Leaving it empty might be better when using legacy pipeline.

Reply:
Does `Builtin` mean legacy pipeline?

Reply:
`Builtin` is the official custom pipeline which is replacing the legacy forward pipeline.

Reply:
This code is saying "If custom pipeline is enabled but `CUSTOM_PIPELINE_NAME` is not set, set it to default 'Builtin'".

Reply:
Got it.